### ISSUE TITLE
TransitionMotion's defaultStyles don't seem to propagate properly

### ISSUE THREAD

**Issue by frekw** [chenglou/react-motion#296]
I'm having some issues when trying to create a staggered animation on render.

Given this example:
```javascript
React.createClass({
  getDefaultStyles(){
    return [/* ... */]
  },

  getStyles(prev) {
    // prev will be undefined here!
  },

  render(){
    <TransitionMotion defaultStyles={this.getDefaultStyles()} styles={this.getStyles}>
    // ...
    </TransitionMotion>
  }
})
```

With this code, `getStyles` will be called with `prev` set to `undefined` the first call, which seems to go against the documentation. If I change the `TransitionMotion` to a `StaggeredMotion` the `prev` value seems to get set as intended during the first call to `getStyles`.

My hunch is that the fix should be to pass the `defaultStyles` to the `styles()` call at https://github.com/chenglou/react-motion/blob/master/src/TransitionMotion.js#L219, but doing seems to badly break the tests, and I'm not familiar enough with the test suite to be confident that I'm not breaking other expected behaviour...

**Comment by PhilipDavis** [comment]
I've worked around it for now by doing:
styles={ (prevStyles) => !prevStyles ? initialStyles : prevStyles.map((style, i) => {
// ...
}) }

**Comment by julienvincent** [comment]
I'm having trouble with this as well and your solutions seems a bit hacky, so I guess this is a bug?

**Comment by chenglou** [comment]
Indeed a bug; thanks for catching it! Fixed and released as v0.4.3
